### PR TITLE
document `finally` in `try/catch`

### PR DESCRIPTION
**Pull request by stevengj**
As noted [on discourse](https://discourse.julialang.org/t/try-catch-online-help/113532), the `try` docstring should at least briefly mention `finally`.

### CODE REVIEW

**Review by Seelengrab — APPROVED** [review]
LGTM

**Comment by jariji** [comment]
This PR copies from the `finally` docstring. In both places it shows how to combine `try` with `finally` but not how to combine `try`/`catch` with `finally`. Eg if you've never seen `finally` before, you might wonder is it try/finally/catch or try/catch/finally or what. Suggested fix: include a try/catch/finally example.

**Comment by stevengj** [comment]
I added a `catch` block to the example.

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -1057,10 +1057,13 @@ used to guarantee that an opened file is closed:
 f = open("file")
 try
     operate_on_file(f)
+catch
+    println("Zoinks!  An error occurred!)
```
Comment (Seelengrab) [inline base/docs/basedocs.jl:1061]:
```suggestion
    println("Zoinks!  An error occurred!")
```

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -1058,7 +1058,7 @@ f = open("file")
 try
     operate_on_file(f)
 catch
-    println("Zoinks!  An error occurred!)
+    println("Zoinks!  An error occurred!")
```
Comment (kimikage) [inline base/docs/basedocs.jl:1061]: "Zoinks!" may be a bit of humor, but is it necessary for docstrings?

Comment (stevengj) [reply base/docs/basedocs.jl:1061]: Yes.  Yes it is.

Comment (kimikage) [reply base/docs/basedocs.jl:1061]: Just to be clear, I am not saying that humor is not needed.
The example above this uses `@warn` to display a simple message. Here, `println` is used to tease.
As a humor deficient person, I think it is better not to require specific cultural knowledge or meta-context to the reader.

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -1049,13 +1049,28 @@ exception object to the given variable within the `catch` block.
...
+
+A `try` or `try`/`catch` block can also have a [`finally`](@ref) clause that executes
+at the end, regardless of whether an exception occurred.  For example, this can be
+used to guarantee that an opened file is closed:
+```julia
+f = open("file")
+try
+    operate_on_file(f)
+catch
+    println("An error occurred!")
```
Comment (stevengj) [inline base/docs/basedocs.jl:1061]:
```suggestion
    @warn "An error occurred!"
```
since the earlier example used `@warn`

**Review by fingolfin — APPROVED** [review]

**Pull request merged**